types/transducers-js/index.d.ts

```diff
@@ -13,10 +14,12 @@ export type Reducer<TResult, TInput> = (result: TResult, input: TInput) => TResu
 
 // Common case: Transducer<TInput, TOutput> =
 //   Transformer<TResult, TOutput> => Transformer<TResult, TInput>.
-export type Transducer<TInput, TOutput> =
-    <TResult, TCompleteResult>(
-        xf: CompletingTransformer<TResult, TCompleteResult, TOutput>
-    ) => CompletingTransformer<TResult, TCompleteResult, TInput>;
+export interface Transducer<TInput, TOutput> {
+    <TResult>(xf: Transformer<TResult, TOutput>)
```

Comment:
I'm not sure about this... the comment above indicates this should be allowed, and without it there's an error in the test... (but it's not directly related).

Reply:
Oh huh, I just checked out the code and you're right about a seemingly unrelated failure.

It looks like that line in the tests
```ts
const xf = t.comp(t.map(reverse), t.cat);
```
is a bit too much to ask from type inference. Without the overload, `xf` is inferred as `CompletingTransformer<{}, {}, number[]>`. Even with the overload which makes the error go a way, `xf` is inferred as the somewhat unsatisfying `Transducer<number[], any>`. I tried to make it work with stuff like
```ts
const xf = t.comp<number[], number[], number>(t.map(reverse), t.cat);
```
which led to other complicated errors.

It looks like this is indicative of a deeper problem that typical usage of this library simply asks too much of the type inferencer. I'm happy to add the overload, and we may just have to live with not getting satisfying types from this library when using `comp`.

(Shameless plug: these sorts of issues are exactly why I wrote [Transducist](https://github.com/dphilipson/transducist)).

Reply:
Yeah, I figured something like that... https://github.com/DefinitelyTyped/DefinitelyTyped/issues/19276#issuecomment-325168002

Thanks for taking a look in any case!

Your lib looks useful! (I'm more a fan of inherent TS libs anyways :-D)